### PR TITLE
[SwiftASTContext] One variant of `GetIdentifier` is enough.

### PR DESCRIPTION
**Pull request by dcci**

### CODE REVIEW

**Comment by dcci** [comment]
@swift-ci test and merge

**Review by bulbazord — APPROVED** [review]

File: source/Symbol/SwiftASTContext.cpp [hunk 1]
```diff
@@ -4548,7 +4542,7 @@ size_t SwiftASTContext::FindTypesOrDecls(const char *name,
   if (name && name[0] && swift_module) {
     swift::ModuleDecl::AccessPathTy access_path;
     llvm::SmallVector<swift::ValueDecl *, 4> value_decls;
-    swift::Identifier identifier(GetIdentifier(name));
+    swift::Identifier identifier(GetIdentifier(llvm::StringRef(name)));
```
Comment (adrian-prantl) [inline source/Symbol/SwiftASTContext.cpp:4545]: We really should also stringref-ify this function...

**Review by adrian-prantl — APPROVED** [review]

**Comment by dcci** [comment]
@swift-ci test osx platform

**Comment by dcci** [comment]
@swift-ci test macos platform

**Pull request merged**